wp-content/themes/chrisdinh-photography/resources/scripts/util/filterGallery.js

```diff
@@ -0,0 +1,53 @@
+import { gsap } from "gsap";
+import { Flip } from "gsap/dist/Flip.js";
+
+// https://gsap.com/community/forums/topic/30406-animate-growing-grid-items/
+// https://codepen.io/GreenSock/pen/mdPzJKp
+
+gsap.registerPlugin(Flip);
+
+const filterGallery = () => {
+  const allFilter = document.querySelector('button.gallery-collections__collection--all');
+  const filters = gsap.utils.toArray('.gallery-collections__collection:not(.gallery-collections__collection--all)');
+  const items = gsap.utils.toArray('.gallery-item__image');
+
+  if ( filters.length ) {
+    filters.forEach(btn => btn.addEventListener('click', () => {
+      updateFilters(filters, items, false);
+    }));
+  }
+
+  if ( allFilter ) {
+    allFilter.addEventListener('click', () => {
+      updateFilters(filters, items, true);
+    });
+  }
+}
+
+const updateFilters = (filters, items, showAll = false) => {
+  const state = Flip.getState(items); // get the current state
+  let collections = [];
+
+  if ( showAll ) {
+    collections = filters.map(filter => filter.dataset.termSlug);
+  } else {
+    collections = filters.filter(filter => filter.classList.contains('gallery-collections__collection--active')).map(item => item.dataset.termSlug);
+  }
+
+  // adjust the display property of each item ("none" for filtered ones, "inline-flex" for matching ones)
+  // items.forEach(item => item.style.display = matches.indexOf(item.dataset.terms) === -1 ? "none" : "inline-flex"); // If we need to support multiple collections on one
+  items.forEach(item => item.style.display = collections.indexOf(item.dataset.terms) === -1 ? "none" : "inline-flex");
```

Comment:
Approach might not work if an item has multiple collections (possibly something to test).

i.e.: if an image has collections: "nature, landscape" this might look for a filter that's called "nature, landscape" instead of filters called "nature" and "landscape".

Reply:
This file is likely stale since we're using nanoGallery to handle filtering currently - i may remove this file after i review the code again.

I _believe_ the way this worked was each gallery item had their collections in a string (`nature,landscape`). The code in this comment looks at the single term selected in the filters and checks if the image has that term name in that string (aka does `nature` exist within `nature,landscape`) in order to filter them